ldclient.go

```diff
@@ -104,9 +105,21 @@ func MakeCustomClient(apiKey string, config Config) LDClient {
 	var streamProcessor *StreamProcessor
 
 	config.BaseUri = strings.TrimRight(config.BaseUri, "/")
-	httpClient := httpcache.NewMemoryCacheTransport().Client()
-	// Client Transport of type *httpcache.Transport doesn't support CancelRequest; Timeout not supported
-	// httpClient.Timeout = config.Timeout
+
+	baseTransport := httpcontrol.Transport{
+		RequestTimeout: config.Timeout,
+		DialTimeout:    config.Timeout,
+		DialKeepAlive:  2 * time.Minute,
```

Comment:
I'm pretty sure this is longer than fastly (or haproxy) will allow a connection to be help open.